extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeStreamExtractor.java

```diff
@@ -193,74 +194,224 @@ public String getSubChannelAvatarUrl() {
+                            id + "-" + DeliveryMethod.HLS,
+                            hlsStreamUrl,
+                            true,
+                            format,
+                            DeliveryMethod.HLS,
+                            UNKNOWN_BITRATE));
+                }
+            }
+            return audioStreams;
+        } catch (final Exception e) {
+            throw new ParsingException("Could not get audio streams from array", e);
+        }
     }
 
     @Override
     public List<VideoStream> getVideoStreams() throws ExtractionException {
         assertPageFetched();
+
         final List<VideoStream> videoStreams = new ArrayList<>();
-        // mp4
-        try {
-            videoStreams.addAll(getVideoStreamsFromArray(json.getArray("files")));
-        } catch (Exception ignored) { }
-        // HLS
-        try {
-            final JsonArray streamingPlaylists = json.getArray("streamingPlaylists");
-            for (final Object p : streamingPlaylists) {
-                if (!(p instanceof JsonObject)) continue;
-                final JsonObject playlist = (JsonObject) p;
-                videoStreams.addAll(getVideoStreamsFromArray(playlist.getArray("files")));
+        if (getStreamType() == StreamType.VIDEO_STREAM) {
+            // Non-HLS streams
+            try {
+                videoStreams.addAll(getVideoStreamsFromArray(json.getArray("files"), ""));
+            } catch (final Exception ignored) {
+            }
+            // HLS streams
+            try {
+                final JsonArray streamingPlaylists = json.getArray("streamingPlaylists");
+                for (final Object p : streamingPlaylists) {
+                    if (!(p instanceof JsonObject)) {
+                        continue;
+                    }
+                    final JsonObject playlist = (JsonObject) p;
+                    final String playlistUrl = playlist.getString("playlistUrl");
+                    videoStreams.addAll(getVideoStreamsFromArray(playlist.getArray("files"),
+                            playlistUrl));
+                }
+            } catch (final Exception e) {
+                throw new ParsingException("Could not get video streams", e);
+            }
+        } else {
+            try {
+                final JsonArray streamingPlaylists = json.getArray("streamingPlaylists");
+                for (final Object p : streamingPlaylists) {
+                    if (!(p instanceof JsonObject)) {
+                        continue;
+                    }
+                    final JsonObject playlist = (JsonObject) p;
+                    videoStreams.add(new VideoStream(
+                            String.valueOf(playlist.getInt("id", 0)),
+                            playlist.getString("playlistUrl", EMPTY_STRING),
+                            true,
+                            MediaFormat.MPEG_4,
+                            DeliveryMethod.HLS,
+                            "",
```

Comment:
Is the resolution unknown?

Reply:
For livestreams, yes, because there may have several streams in the HLS playlist URL.

Reply:
In this case, please add some kind of documentation for this to `VideoStream.getResolution()` method.  I am not expecting an empty String as resolution. I think it is good to state that the resolution is unkown when the DeliveryMethod is HLS. 
Please add a new constant for the unknown stream resolution.

Reply:
What constant should I use when the resolution is unknwon? An empty string or something else?

Reply:
that's ok

Reply:
Done